lib/blueprinter/field_map.rb

```diff
@@ -0,0 +1,64 @@
+module Blueprinter
+  # @api private
+
+  class FieldMap
+
+    def initialize
+      @ancestors = {
+        identifier: [:identifier], default: [:identifier, :default]
+      }
+      @fields = { identifier: [], default: [] }
+      @excluded_fields = {}
+      @composed_fields = {}
+    end
+
+    def include_view(view, ancestor_view = :default)
```

Comment:
NB: 

(This PR looks good as it is, just a refactor proposition for future. Please feel free to completely disregard my comment 😄)

Not very familiar with the internals of Blueprinter, but I think it would be great if we could work towards separating out the methods `include_view` from `add_field` and `exclude_field`.

Ideally, if we had:
- `FieldMap` - does the basic unit of work, of adding/excluding attributes from a view using `add_field` and `exclude_field`
- `ViewBuilder` -  utilizes the `FieldMap` and builds a view, also capable of recursively building `included views`
- `ViewComposer` - composes a view based on the attributes resolved by `ViewBuilder`

`Blueprint.render` should then be able to render view based on fields composed by `ViewComposer`.